custom_components/hubspace/light.py

```diff
@@ -563,86 +358,61 @@ def brightness(self) -> int or None:
-        """Return the warmest color_temp that this light supports."""
-        return self._max_mireds
-
     @property
     def is_on(self) -> bool | None:
         """Return true if light is on."""
         if self._state is None:
             return None
-        else:    
+        else:
             return self._state == "on"
 
+    @property
+    def min_color_temp_kelvin(self) -> int:
+        return min(self._temperature_choices)
+
+    @property
+    def max_color_temp_kelvin(self) -> int:
+        return max(self._temperature_choices)
+
     def send_command(self, field_name, field_state, functionInstance=None) -> None:
         self._hs.setState(self._childId, field_name, field_state, functionInstance)
 
     def set_send_state(self, field_name, field_state) -> None:
         self._hs.setState(self._childId, field_name, field_state)
 
     def turn_on(self, **kwargs: Any) -> None:
-        state = self._hs.setPowerState(
-            self._childId, "on", self._usePowerFunctionInstance
-        )
-
-        if ATTR_BRIGHTNESS in kwargs and (
-            ColorMode.ONOFF not in self._supported_color_modes
-        ):
+        """Perform power on and set additional attributes"""
+        _LOGGER.debug(f"Adjusting light {self._childId} with {kwargs}")
+        power_state = {
+            "functionClass": "power",
+            "value": "on",
+        }
+        if self._usePowerFunctionInstance:
+            power_state["functionInstance"] = self._usePowerFunctionInstance
+        states_to_set = [power_state]
+        if ATTR_BRIGHTNESS in kwargs and ColorMode.BRIGHTNESS in self._supported_color_modes:
             brightness = kwargs.get(ATTR_BRIGHTNESS, self._brightness)
-            self._hs.setState(
-                self._childId, "brightness", _brightness_to_hubspace(brightness)
+            states_to_set.append(
+                {
+                    "functionClass": "brightness",
+                    "value": _brightness_to_hubspace(brightness),
+                }
             )
-
-        if ATTR_RGB_COLOR in kwargs and any(
-            mode in COLOR_MODES_COLOR for mode in self._supported_color_modes
-        ):
-            self._hs.setRGB(self._childId, *kwargs[ATTR_RGB_COLOR])
-
-        if ATTR_WHITE in kwargs and (
-            any(mode in COLOR_MODES_COLOR for mode in self._supported_color_modes)
-            or ColorMode.COLOR_TEMP in self._supported_color_modes
-        ):
-            self._colorMode = ATTR_WHITE
-            self._hs.setState(self._childId, "color-mode", self._colorMode)
-            brightness = kwargs.get(ATTR_WHITE, self._brightness)
-            self._hs.setState(
-                self._childId, "brightness", _brightness_to_hubspace(brightness)
+        if ATTR_COLOR_TEMP in kwargs and ColorMode.COLOR_TEMP in self._supported_color_modes:
+            color_to_set = self._temperature_choices[0]
+            # I am not sure how to set specific values, so find the value
```

Comment:
See note. If theres an easy way to provide this information, id prefer that route than having to calculate it each time